website/main/content/blog/2025-05-26-longest-chain-rule.md

```diff
@@ -0,0 +1,169 @@
+  that gives enough time to consider the segments final and to challenge potential misbehaviors and
+  mistakes.
+
+To try and clarify the description above, let me share how this would work step-by-step for the
+commitment of a segment `s1` for a leaf segment:
+
+1. A farmer in the leaf shard creates a segment `s1` that is included and submitted to the parent as
+   part of the `own_segments_root` field of the `LeafShardBlockInformation` data structure that is
+   included in `blkA` of its intermediate shard parent. `s1` was created in `blk1` of the leaf
+   shard.
+2. As soon as the parent sees this new child segment, it includes it in its next
+   `IntermediateShardBlockInformation` inside `child_segment_root` included in `blkX` of the beacon
+   chain. This makes the leaf segment immediately available in the beacon chain, but it won't be
+   included into the global history, and be included in the corresponding super segment, until the
+   probability of re-org of all the blocks involved in the submission of the segment, mainly `blk1`,
+   `blkA` and `blkX` is over the finality threshold chosen.
+3. The intermediate shard is periodically updating its perceived probability of reorg for the leaf's
+   shard `blk1` and its will notify the beacon chain in the following
+   `IntermediateShardBlockInformation` that the finality threshold has been achieved. Similarly, the
+   beacon chain keep track of this same threshold for `blkA`. When both are confirmed as final, the
+   leaf segment of that block is committed as part of the global history, assigned a global piece
+   index within that history, and made available for archiving.
+
+## Computing the block probability of reorganisation
+
+And many of you may be wondering at this point, but how can we compute the probability of
+reorganisation for a block in the first place? Fortunately, there is a lot of literature around this
+topic and attempts to objectively measure this metric or related ones for longest-chain protocols
+(including the original Bitcoin white paper, with its analysis of its probability of attack):
+
+- [Satoshi Nakamoto's Bitcoin Whitepaper](https://bitcoin.org/bitcoin.pdf)
+- ["On Finality in Blockchains" by Anceaume et al.](https://arxiv.org/abs/2012.10172)
+- And quantitative approaches to blockchain finality for
+  [Markov Chain models](https://link.springer.com/article/10.1007/s00145-016-0259-8), and formal
+  analysis of Bitcoin's backbone and for
+  [game theory and selfish mining](https://www.cs.cornell.edu/~ie53/publications/selfish-mining.pdf)
+  (relevant to miner incentives and reorgs).
+
+From the analyses on these papers, I extracted the following high-level formula to compute the
+probability of block reorganisation that we will be able to combine to our needs:
+
+```rust
+Prob (num_blocks_replaced) = (2*time_to_first_block_propagation/avg_time_between_blocks)^(num_blocks_replaced)
+```
```

Comment:
Actual formulas can be enabled here too if you want: https://gohugo.io/content-management/mathematics/